cachi2/core/package_managers/yarn_classic/project.py

```diff
@@ -0,0 +1,294 @@
+                solution=(
+                    "The package.json file must contain valid JSON. "
+                    "Refer to the parser error and fix the contents of the file."
+                ),
+            )
+
+        return cls(path, package_json_data)
+
+
+class YarnRc(_CommonConfigFile):
+    """A yarnrc file."""
+
+    _cache_folder: Optional[str]
+
+    @property
+    def cache_folder(self) -> str:
+        """Get the configured location for the yarn cache folder.
+
+        Fallback to the default path in case the configuration key is missing.
+        """
+        return self._data.get("--cache-folder", DEFAULT_CACHE_FOLDER)
+
+    @classmethod
+    def from_file(cls, path: RootedPath) -> "YarnRc":
+        """Parse the content of a yarnrc file.
+
+        .yarnrc file are seriously dumb, just plain text keys and values with a
+        single space in between. According to
+        https://classic.yarnpkg.com/en/docs/yarnrc, paths should be quoted
+
+        `yarn-offline-mirror "./packages-cache"`
+
+        CLI syntax is possible
+
+        `--install.check-files true`
+
+        but paths don't have to be quoted(?!)
+
+        `--cache-folder /tmp/yarn-cache/`
+        """
+
+        yarnrc_data = {}
+        try:
+            with path.path.open("r") as f:
+                for line in f:
+                    line = line.replace("\n", "").strip()
+                    if line:
+                        key_and_val = line.split(" ", 1)
+                        yarnrc_data[key_and_val[0]] = key_and_val[1]
+        except ValueError as e:
+            raise PackageRejected(
+                reason=f"Can't parse the {path.subpath_from_root} file. Parser error: {e}",
+                solution=(
+                    "The yarnrc file must contain valid data."
+                    "Refer to the parser error and fix the contents of the file."
+                ),
+            )
+
+        if not yarnrc_data:
+            # warn
+            pass
+
+        return cls(path, yarnrc_data)
+
+    @classmethod
+    def from_str(cls, string: str) -> dict[str, Any]:
+        """Parse the content of a string containing the contents of a valid yarnrc file."""
+        yarnrc_data = {}
+        try:
+            for line in string.splitlines():
+                line.strip()
+                if line:
+                    key_and_val = line.split(" ", 1)
+                    yarnrc_data[key_and_val[0]] = key_and_val[1]
+        except ValueError as e:
+            raise PackageRejected(
+                reason=f"Can't parse the string. Parser error: {e}",
+                solution=(
+                    "The string must contain valid yarnrc data."
+                    "Refer to the parser error and fix the contents of the string."
+                ),
+            )
+
+        if not yarnrc_data:
+            # warn
+            pass
+
+        return yarnrc_data
+
+    def write(self) -> None:
```

Comment:
'.yarnrc' and 'package.json' both have `write()` methods in cachi2's Yarnberry implementation.

Reply:
> Why would we want to write to any of these files? Should not they be read-only for us?

We actually do write them in the temp clean source repo clone when fetching & validating to disable some default Yarn functionality. I don't remember whether everything could have been set with env variables, but we do write the config actually before interacting with Yarn V2 since users are able to ship their own Yarn shim in their repo. IIRC plugins can't be disabled/set via env variables and so they have to be overridden via `.yarnrc` and so I presume we decide to write the config instead of doing both env vars AND the config.

Reply:
@brunoapimentel wrote that specific bit of code for Yarn v2 so his memory is probably better than mine on that matter.

Reply:
In the **_initial implementation_** for yarn 1.x support, we shouldn't need to write these files, but that may change in the future with follow-ups

Reply:
Removed all `writes()`